sdk/formrecognizer/azure-ai-formrecognizer/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,244 @@
+with open(path_to_sample_documents, "rb") as f:
+    poller = document_analysis_client.begin_analyze_document(
+        "prebuilt-layout", document=f
+    )
+result = poller.result()
+
+for idx, style in enumerate(result.styles):
+    # process document styles
+
+for page in result.pages:
+    # process pages
+    for line in page.lines:
+        # process lines
+    for word in page.words:
+        # process words
+
+for table_idx, table in enumerate(result.tables):
+    # process tables found in document
+```
+
+Analyzing general prebuilt document types with `3.2.x`:
+```python
+with open(path_to_sample_documents, "rb") as f:
+    poller = document_analysis_client.begin_analyze_document(
+        "prebuilt-document", document=f
+    )
+result = poller.result()
+
+for entity in result.entities:
+    # process entities found in document
+
+for kv_pair in result.key_value_pairs:
+    # process key-value pairs found in document
+
+for page in result.pages:
+    # process pages
+    for line in page.lines:
+        # process lines
+    for word in page.words:
+        # process words
+
+for table_idx, table in enumerate(result.tables):
+    # process tables found in document
+```
+
+NOTE: All of these samples also work with `begin_analyze_document_from_url` when providing a valid URL to the document.
+
+### Analyzing a custom model
+
+Differences between the versions:
+- Analyzing a custom model with `DocumentAnalysisClient` uses the general `begin_analyze_document` and `begin_analyze_document_from_url` methods.
+- In order to analyze a custom model with `FormRecognizerClient` the `begin_recognize_custom_models` and its corresponding URL methods are used.
+- The `include_field_elements` kwarg does not exist with the `DocumentAnalysisClient`, text details are automatically included with service version `2021-09-30-preview`.
+
+Analyze custom document with `3.1.x`:
+```python
+with open(path_to_sample_forms, "rb") as f:
+    poller = form_recognizer_client.begin_recognize_custom_forms(
+        model_id=model_id, form=f, include_field_elements=True
+    )
+forms = poller.result()
+```
+
+Analyze custom document with `3.2.x`:
+```python
+with open(path_to_sample_documents, "rb") as f:
+    poller = document_analysis_client.begin_analyze_document(
+        model=model_id, document=f
+    )
+result = poller.result()
+```
+
+### Training a custom model
+
+Differences between the versions:
+- Files for building a new model for version `3.2.x` can be created using the new labeling tool found [here](TODO).
+- In version `3.1.x` the `use_training_labels` kwarg was used to indicate whether custom labeling was when creating the new models.
+- In version `3.2.x` the `use_training_labels` kwargs is not supported since training must be carried out with labeled training documents. In order to extract key-value pairs from a document, please refer to the prebuilt model "prebuilt-document" which extracts entities, key-value pairs, and layout from a document. 
```

Comment:
Since we didn't expose key-value pairs as key value pairs in v3.1.x, you might want to describe how prebuilt-document satisfies the training without labels use case + the new stuff it returns.